tests/suites/test_suite_ssl.function

```diff
@@ -3184,6 +3192,42 @@ void handshake( const char *cipher, int version, int pk_alg,
     TEST_ASSERT( client.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
     TEST_ASSERT( server.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
 
+#if defined(MBEDTLS_SSL_CONTEXT_SERIALIZATION)
+    if( dtls == 0 || serialize == 0 )
+        goto exit;
+
+    memset( cli_buf, 'X', MSGLEN );
+    memset( srv_buf, 'Y', MSGLEN );
+
+    /* Make sure that writing/reading works */
+    TEST_ASSERT( mbedtls_ssl_write( &(client.ssl), cli_buf, MSGLEN ) == MSGLEN );
+    TEST_ASSERT( mbedtls_ssl_read( &(server.ssl), srv_buf, MSGLEN ) == MSGLEN );
+    TEST_ASSERT( memcmp( cli_buf, srv_buf, MSGLEN ) == 0 );
+
+    TEST_ASSERT( mbedtls_ssl_context_save( &(server.ssl), NULL,
+                                           0, &context_buf_len )
+                                         == MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );
+
+    context_buf = mbedtls_calloc( 1, context_buf_len );
+    TEST_ASSERT( context_buf != NULL );
+
+    TEST_ASSERT( mbedtls_ssl_context_save( &(server.ssl), context_buf,
+                                          context_buf_len,
+                                          &context_buf_len ) == 0 );
+
+    TEST_ASSERT( mbedtls_ssl_context_load( &(server.ssl), context_buf,
+                                           context_buf_len ) == 0 );
```

Comment:
It would not be better if we do for example `mbedtls_ssl_free` between  `mbedtls_ssl_context_save` and `mbedtls_ssl_context_load` to make sure serialization works? Or use `mbedtls_ssl_context_load` on another object?

Reply:
There are two layers of serialization used in `ssl_server2` serialization. This is the first one, and the second one  adds additional `mbedtls_ssl_free`, `mbedtls_ssl_init` followed by a bunch of setup stuff. I can add this too if we want to.

Reply:
Could we memset the buffer to all zeros in-between saving and loading at least? I'd like to have more confidence that the saved context doesn't have any old data in it that gets reused.